src/Hyperion/Main.hs

```diff
@@ -78,7 +81,8 @@ options = do
          (Options.strOption
             (Options.long "output" <>
              Options.short 'o' <>
-             Options.metavar "FILENAME"))
+             Options.help "Outputs the benchark to the directory" <>
```

Comment:
Th reason why we wanted to have a directory in the first place is to deal with the fact that the same arguments are passed to all benchmarks executable (defined in `.cabal`), thus providing a file would have them overwrite each other, as each benchmark executable would try to write to the same file.

The `hasTrailingPathSeparator` indeed sounds like the proper solution because it would allow a user to output its content to a folder without having to create it beforehand.

Having a predefined naming convention also seemed like a win because it makes parsing easier and allows an simple output name for raw output (packageName.executableName.raw.json)

Does the following behaviour suit you:
- If output is a directory that exists, output to the directory with the naming convention
- else
  - if `hasTrailingPathSeparator`, create the directory and write to it
  - otherwise output to teh file and create it if needed.